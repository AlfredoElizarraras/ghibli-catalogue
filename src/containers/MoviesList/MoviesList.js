import React from 'react';
import PropTypes from 'prop-types';
import Movie from '../../components/Movie';
import Filter from '../../components/Filter';
import styles from './moviesList.module.css';

const MoviesList = ({
  movies = [], directors, producers, changeFilter,
}) => {
  const handleFilterChange = filter => {
    const { name, value } = filter;
    changeFilter({ name, value });
  };

  return (
    <>
      <div className={styles.filters}>
        <Filter type="director" handleFilterChange={handleFilterChange} data={directors} />
        <Filter type="producer" handleFilterChange={handleFilterChange} data={producers} />
      </div>
      <div className={styles.movies}>
        {
          movies.map(movie => (
            <Movie key={movie.id} movie={movie} />
          ))
        }
      </div>
    </>
  );
};

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      director: PropTypes.string.isRequired,
      producer: PropTypes.string.isRequired,
      release_date: PropTypes.string.isRequired,
      rt_score: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    }),
  ).isRequired,
  directors: PropTypes.arrayOf(
    PropTypes.string,
  ).isRequired,
  producers: PropTypes.arrayOf(
    PropTypes.string,
  ).isRequired,
  changeFilter: PropTypes.func.isRequired,
};

export default MoviesList;
