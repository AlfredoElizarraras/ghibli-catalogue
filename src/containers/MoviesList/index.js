import { connect } from 'react-redux';
import MovieList from './MoviesList';
import changeFilter from '../../redux/actions/filtersActions';
import { getMoviesByFilter, getAllOfProperty } from '../../redux/selectors/selectors';

const mapStateToProps = state => {
  const { filters } = state;
  const movies = getMoviesByFilter(state, { filter: 'director', value: filters });
  const directors = getAllOfProperty(state, 'director');

  return { movies, directors };
};

export default connect(mapStateToProps, { changeFilter })(MovieList);
