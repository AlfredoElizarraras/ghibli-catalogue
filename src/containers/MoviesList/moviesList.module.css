.filters {
  display: flex;
  align-items: center;
}

.movies {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
}
