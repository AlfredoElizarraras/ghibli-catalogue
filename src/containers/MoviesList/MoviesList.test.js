import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MoviesList from './MoviesList';

describe('MoviesList', () => {
  test('should render without crashing', () => {
    const props = {
      movies: [
        {
          id: '1',
          title: 'Movie 1',
          director: 'director 1',
          description: 'Movie description',
          producer: 'Movie producer',
          release_date: 'Movie release date',
          rt_score: 'Movie rate score',
          url: 'Movie URL',
        },
        {
          id: '2',
          title: 'Movie 2',
          director: 'director 2',
          description: 'Movie description',
          producer: 'Movie producer',
          release_date: 'Movie release date',
          rt_score: 'Movie rate score',
          url: 'Movie URL',
        },
        {
          id: '3',
          title: 'Movie 3',
          director: 'director 3',
          description: 'Movie description',
          producer: 'Movie producer',
          release_date: 'Movie release date',
          rt_score: 'Movie rate score',
          url: 'Movie URL',
        },
        {
          id: '4',
          title: 'Movie 4',
          director: 'director 3',
          description: 'Movie description',
          producer: 'Movie producer',
          release_date: 'Movie release date',
          rt_score: 'Movie rate score',
          url: 'Movie URL',
        },
      ],
      directors: ['director 1', 'director 2', 'director 3'],
      changeFilter: filter => filter,
    };

    const tree = renderer.create(
      <BrowserRouter>
        <MoviesList
          movies={props.movies}
          directors={props.directors}
          changeFilter={props.changeFilter}
        />
      </BrowserRouter>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
