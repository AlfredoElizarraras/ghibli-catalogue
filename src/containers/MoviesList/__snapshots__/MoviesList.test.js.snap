// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render without crashing 1`] = `
Array [
  <div
    className="filters"
  >
    <label
      className="filter"
      htmlFor="director"
    >
      Fiter by
       
      director
      <select
        id="director"
        name="director"
        onChange={[Function]}
        type="director"
        value="ALL"
      >
        <option>
          ALL
        </option>
        <option
          value="director 1"
        >
          DIRECTOR 1
        </option>
        <option
          value="director 2"
        >
          DIRECTOR 2
        </option>
        <option
          value="director 3"
        >
          DIRECTOR 3
        </option>
      </select>
    </label>
  </div>,
  <div
    className="movies"
  >
    <div
      className="movie"
    >
      <a
        className="imageContainer"
        href="/1"
        onClick={[Function]}
      >
        <img
          alt="Totoro"
        />
        Movie 1
      </a>
      <span>
        director 1
      </span>
    </div>
    <div
      className="movie"
    >
      <a
        className="imageContainer"
        href="/2"
        onClick={[Function]}
      >
        <img
          alt="Totoro"
        />
        Movie 2
      </a>
      <span>
        director 2
      </span>
    </div>
    <div
      className="movie"
    >
      <a
        className="imageContainer"
        href="/3"
        onClick={[Function]}
      >
        <img
          alt="Totoro"
        />
        Movie 3
      </a>
      <span>
        director 3
      </span>
    </div>
    <div
      className="movie"
    >
      <a
        className="imageContainer"
        href="/4"
        onClick={[Function]}
      >
        <img
          alt="Totoro"
        />
        Movie 4
      </a>
      <span>
        director 3
      </span>
    </div>
  </div>,
]
`;
